### ISSUE TITLE
API to change a tab's windowId: tab.move()

### ISSUE THREAD

**Issue by petemill** [brave/muon#489]
At the moment, in order to move a tab to a new window, you must perform the following steps:
1. `tab.detach()`
2. Attach the tab's guestInstance in a different window via a `<webview>` element
3. `tab.setTabIndex(newIndex)`

It would be ideal to have a method similar to [Web Extension's `tab.move(tabId, { windowId, index})`](https://developer.chrome.com/extensions/tabs#method-move).

This method would internal change the `windowId` and `index` associated with the tab, without having to detach / attach the tab to a `<webview>`.
